### PR TITLE
Minor revisions to ```simulation.py```

### PR DESCRIPTION
**Pull request by David-Woroniuk**
- Typechecks
- Type Annotation

Removed：

- Non CamelCase Class Names (PEP:8)
- CamelCase local varibales within functions (should be snakecase PEP:8)

### CODE REVIEW

File: pycop/simulation.py [hunk 1]
```diff
@@ -24,19 +26,22 @@ def simu_gaussian(n, m, corrMatrix):
         the simulated sample 
 
     """
-
+    if not all(isinstance(v, int) for v in [n, m]):
+        raise TypeError("The 'n' and 'm' arguments must both be integer types.")
+    if not isinstance(corr_matrix, np.array):
```
Comment (maximenc) [inline pycop/simulation.py:31]: Seems that for checking the type we need provide "np.ndarray" and not "np.array":
```suggestion
    if not isinstance(corr_matrix, np.ndarray):
```

File: pycop/simulation.py [hunk 2]
```diff
@@ -57,13 +62,19 @@ def simu_tstudent(n, m, corrMatrix, nu):
         the simulated sample 
 
     """
+    if not all(isinstance(v, int) for v in [n, m]):
+        raise TypeError("The 'n' and 'm' arguments must both be integer types.")
+    if not isinstance(corr_matrix, np.array):
```
Comment (maximenc) [inline pycop/simulation.py:67]:
```suggestion
    if not isinstance(corr_matrix, np.ndarray):
```

File: pycop/simulation.py [hunk 3]
```diff
@@ -57,13 +62,19 @@ def simu_tstudent(n, m, corrMatrix, nu):
...
+        raise TypeError("The 'corr_matrix' argument must be a numpy array.")
+    if not isinstance(nu, float):
```
Comment (maximenc) [inline pycop/simulation.py:69]: I would consider the possibility that the variable is passed as an integer in the following:
```suggestion
    if not isinstance(nu, (int, float)):
```
For the typing, it seems that we can use _float_ only, as _int_ is implied in that type.

File: pycop/simulation.py [hunk 4]
```diff
@@ -93,13 +104,17 @@ def SimuSibuya(alpha, m):
         the simulated sample 
 
     """
+    if not isinstance(alpha, float):
```
Comment (maximenc) [inline pycop/simulation.py:107]:
```suggestion
    if not isinstance(alpha, (int, float)):
```

File: pycop/simulation.py [hunk 5]
```diff
@@ -138,60 +152,50 @@ def simu_archimedean(familly, n, m, theta):
         the simulated sample, array matrix with dim (m, n)
 
     """
-
-    if familly == "clayton":
+    if family not in ["clayton", "gumbel", "frank", "joe", "amh"]:
+        raise ValueError("The family argument must be one of 'clayton', 'gumbel', 'frank', 'joe' or 'amh'.")
+    if not all(isinstance(v, int) for v in [n, m]):
+        raise TypeError("The 'n' and 'm' arguments must both be integer types.")
+    if not isinstance(theta, float):
```
Comment (maximenc) [inline pycop/simulation.py:159]:
```suggestion
    if not isinstance(theta, (int, float)):
```

**Review by maximenc — CHANGES_REQUESTED** [review]
Seems that there are type errors when checking the modification in _simulation.py_. The issue is resolved when considering _np.ndarray_ and the possibility of a variable being _int_ or _float_ when testing the type (in _isinstance()_).

Some minor changes would be required in the simulation section of the readme file, especially to change the matrix variable as a numpy array.

**Comment by David-Woroniuk** [comment]
Hey @maximenc - I agree with each of the above, I was unsure of the ```nu``` value as an ```int``` type. I think this should be ready to merge, assuming it passes unit tests.

**Pull request merged**